### PR TITLE
Implemented basic OTP as mentioned in #3 issue

### PR DESCRIPTION
**Pull request by hoomaac**
Main things have been changed are:

- add basic OTP
- refactor some packages
- add login and register requests and responses

### CODE REVIEW

File: pkg/redis/redis.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	if err != nil {
+		log.Printf("set key=%s, value=%v to redis failed due to %v", key, value, err)
+		return false
+	}
+
+	return true
+}
+
+func GetValue(key string) string {
```
Comment (7eRoM) [inline pkg/redis/redis.go:52]: According to `SetValue`, the return type of `GetValue` should be `interface{}`. Shouldn't it?

File: controllers/v1/goods/add.go [hunk 2]
```diff
@@ -4,8 +4,8 @@ import (
 	"net/http"
 
 	"github.com/gin-gonic/gin"
-	"github.com/hoomaac/vertex/common"
 	"github.com/hoomaac/vertex/models"
+	"github.com/hoomaac/vertex/pkg/database"
```
Comment (7eRoM) [inline controllers/v1/goods/add.go:8]: In a sample project, package `database` was placed in `Model`.

File: main.go [hunk 3]
```diff
@@ -4,30 +4,47 @@ import (
...
+	redis.InitRedis(redisConf)
 
 	// migrate each model here
-	common.Migrate(&models.User{})
-	common.Migrate(&models.Good{})
+	database.Migrate(&models.User{})
+	database.Migrate(&models.Good{})
 }
 
 func main() {
```
Comment (7eRoM) [inline main.go:50]: uhmmmm... It seems `jwt` is not added to middleware.

File: pkg/redis/redis.go (hunk 1 above)
Comment (hoomaac) [reply pkg/redis/redis.go:52]: Actually RedisClient.Get returns a result which consists of a string and an error. As long as we're sure that the type of the value is string we can use this function, but other types such as List, Set a different approach is mostly required.

**Pull request merged**